test-data/unit/check-enum.test

```diff
@@ -1677,3 +1677,22 @@ class A(Enum):
     class Inner: pass
 class B(A): pass  # E: Cannot inherit from final class "A"
 [builtins fixtures/bool.pyi]
+
+[case testEnumNotFinalWithMethodsAndUninitializedValues]
+# https://github.com/python/mypy/issues/11578
+from enum import Enum
+from typing import Final
+
+class A(Enum):
+    x: int
+    def method(self) -> int: pass
+
+class B(A):
+    x = 1  # E: Cannot override writable attribute "x" with a final one
```

Comment:
This needs to be fixed. I am working on it right now.

Reply:
But, on the other hand. Looks like it is the same for regular classes:

```python
from typing import Final

class A:
    x: int
    y: int

class B(A):
    x: Final = 1
    y: Final = 2
# out/ex.py:8: error: Cannot override writable attribute "x" with a final one
# out/ex.py:9: error: Cannot override writable attribute "y" with a final one
```

Reply:
It doesn't look like [PEP-591](https://www.python.org/dev/peps/pep-0591/) prohibits `Final` on a subclass, so it looks like it's a logic error that Mypy does not allow subclassing like you did ^^^.

Reply:
Though this is pretty clear why this is the case (and makes sense) https://github.com/python/mypy/blob/d7c4e6908c71a45e6fab73887b3d1a42a7848247/mypy/checker.py#L2505-L2525 (it's just not covered by PEP-591. It may be possible to check if the base class just has a type like you did in the Enum code.

Reply:
Ok, let's make this a subject for a next PR. This would require some extra work. And might have unwanted consequences.

Reply:
Thanks for your help! 👍

Reply:
Overriding a non-final attribute with a final attribute is a Liskov substitution violation. I'd expect that PEP 591 didn't intend to allow it, but it just doesn't go into enough detail to explicitly mention it. cc @ilevkivskyi

Reply:
> Overriding a non-final attribute with a final attribute is a Liskov substitution violation. I'd expect that PEP 591 didn't intend to allow it, but it just doesn't go into enough detail to explicitly mention it. cc @ilevkivskyi

I figured that might be the case.

Reply:
Yeah. This is intended. I don't remember why we didn't mention it in the PEP (it may be we indeed just wanted to be terse).

Reply:
It should still be allowed if the variable is not initialized yet, right? Based on the PEP it looks like it would be expected to be initialized in the `__init__`, but it sounds like there should be a special case where the class is basically completely abstract.

Reply:
No, it shouldn't. Consider this
```python
class A:
    x: int

class B(A):
    x: Final = 1

# somewhere in modules that depend on the above, and therefore are being processed later
def set_x(a: A) -> None:
    a.x = 42

set_x(B())  # ???
```

Reply:
That makes sense in general but I wonder if it makes sense to special case enums here. We may need more details from the use case in #11578.

Reply:
The example given here https://github.com/python/mypy/issues/11578#issuecomment-973993876 is basically how the code is used. The only thing that the base class does is define a `__new__` that stores the actual values that all the subclasses provide. The interpreter doesn't complain about subclassing an enum, and apparently that's because there are no values defined on the base class (I didn't realize enums don't allow subclassing). I do agree enums probably need a special case because the class that _does_ define enum values is the concrete class that is final whereas the example above the interpreter does not effectively define final like it would for enums.